### PR TITLE
fix: 20585 - Web - Split Bill - Money request confirmation list stays highlighted after press

### PR DESCRIPTION
**Pull request by Skalakid**
### Fixed Issues

$ https://github.com/Expensify/App/issues/20585
PROPOSAL: https://github.com/Expensify/App/issues/20585#issuecomment-1593021403

### Tests

1. Log in
2. Create group chat (click floating `+`  button > `New group` > select some users)
3. Enter group chat
4. Click `+` button that is in the `TextInput`
5. Select `Split bill`
6. Enter any amount of money
7. Start selecting and unselecting people in the list, and check if only the first unselected user is highlighted

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests
### QA Steps

Same as Tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/39538890/71c1c969-89d1-4c4b-9970-d05aeabdeaec

</details>

<details>
<summary>Mobile Web - Chrome</summary>

N/A
</details>

<details>
<summary>Mobile Web - Safari</summary>

N/A
</details>

<details>
<summary>Desktop</summary>

N/A
</details>

<details>
<summary>iOS</summary>

N/A
</details>

<details>
<summary>Android</summary>

N/A
</details>

### CODE REVIEW

File: src/components/Pressable/PressableWithFeedback.js [hunk 1]
```diff
@@ -66,17 +66,15 @@ const PressableWithFeedback = forwardRef((props, ref) => {
                     });
                 });
             }}
+            hoverStyle={props.hoverStyle}
+            pressStyle={props.pressStyle}
+            focusStyle={props.focusStyle}
```
[the remark this answers is not in the document]
Comment (Skalakid) [reply src/components/Pressable/PressableWithFeedback.js:71]: When creating these Pressable components I think we wanted to separate styles from GenericPressable and use them only in child component - OpacityView

[the remark this answers is not in the document]
Comment (Skalakid) [reply src/components/Pressable/PressableWithFeedback.js:71]: By passing them, to the parent component, we can easily fix [the bug with blue borders](https://github.com/Expensify/App/pull/19612#issuecomment-1569876479)

[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/Pressable/PressableWithFeedback.js:71]: Hmm I think the current implementation is the correct now, not sure why we did it that way but I think probably to support `Button`. This change breaks `Button` styles.

Didn't dive into the problem but seems like this should be fixed on OptionRow level.

@Skalakid I see you are not assigned on the linked issue yet. Were you assigned somewhere else?

[the remark this answers is not in the document]
Comment (Skalakid) [reply src/components/Pressable/PressableWithFeedback.js:71]: No, I wasn't assigned somewhere else, I've added the fix as it is a regression from my previous PR https://github.com/Expensify/App/pull/19612 and @Christinadobrzyn asked me for some help.

Btw, can you provide more information on why this change brakes `Button` styles, please?

[the remark this answers is not in the document]
Comment (s77rt) [reply src/components/Pressable/PressableWithFeedback.js:71]: The change itself is a breaking change, styles are no longer applied as they were. To reproduce the bug just hover any button e.g. Settings -> Payments -> "Add payment method" button

| Before | After |
|--------|-------|
|        |       |

**Comment by Skalakid** [comment]
@s77rt I've added some changes, these should fix styling issues

**Comment by Skalakid** [comment]
I made changes in `PressableWithFeedback` to don't break the previous [fix of the focus blue borders](https://github.com/Expensify/App/pull/19612#issuecomment-1569876479). The problem is that we can focus on clickable elements using the keyboard's `tab` button. When we select items using `tab`, elements are being focused but with an additional blue border. Previously we just changed focus styles to be the same as hover styles to just hide the blue border. But now we want to delete focusStyle because it causes the bug with highlighting from this issue.
When we just remove `focusStyle` it will cause something like this:

https://github.com/Expensify/App/assets/39538890/7c6bda88-db55-467f-a258-2899bae3cd44

The problem is that we are focussing on pressable but the `OpacityView` styles are above the blue border and a user can't see it. So I changed the order of the components, and now the Pressable that we are focussing on is the child component. So we can see the blue border and all other styles at the same time! 


https://github.com/Expensify/App/assets/39538890/f31df5dd-f3fa-47cf-ad5d-308c31ba5a34

**Comment by s77rt** [comment]
Bug: Buttons not being dimmed on press.

**Review by Julesssss — DISMISSED** [review]
Tested again and I see the button dimming on web now 👍

File: src/components/OpacityView.js [hunk 2]
```diff
@@ -49,11 +47,7 @@ function OpacityView(props) {
         }
     }, [props.shouldDim, props.dimmingValue, opacity]);
 
-    return (
-        <Animated.View style={[opacityStyle]}>
-            <View style={StyleUtils.parseStyleAsArray(props.style)}>{props.children}</View>
-        </Animated.View>
-    );
+    return <Animated.View style={[props.style, opacityStyle]}>{props.children}</Animated.View>;
```
Comment (s77rt) [inline src/components/OpacityView.js:50]: This is against the styling guide. We should still `parseStyleAsArray` and deconstruct the styles.
```js
<Animated.View style={[opacityStyle, ...StyleUtils.parseStyleAsArray(props.style)]}>{props.children}</Animated.View>;
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Review by s77rt — APPROVED** [review]
LGTM! :rocket: 

cc @Julesssss

**Review by Julesssss — APPROVED** [review]

File: src/components/Pressable/PressableWithFeedback.js
Comment (s77rt) [inline src/components/Pressable/PressableWithFeedback.js:1]: @Skalakid Just found out we are doing something wrong here. Can you please raise a quick follow up to fix this.
1. The rest props we pass to `GenericPressable` should be the first not to overwrite our props (onHoverIn, onPress, etc)
2. The omittedProps should only be `wrapperStyle`

Something like that
```js
diff --git a/src/components/Pressable/PressableWithFeedback.js b/src/components/Pressable/PressableWithFeedback.js
index 2949683734..3defd7d640 100644
--- a/src/components/Pressable/PressableWithFeedback.js
+++ b/src/components/Pressable/PressableWithFeedback.js
@@ -7,7 +7,7 @@ import GenericPressablePropTypes from './GenericPressable/PropTypes';
 import OpacityView from '../OpacityView';
 import variables from '../../styles/variables';
 
-const omittedProps = ['wrapperStyle', 'onHoverIn', 'onHoverOut', 'onPressIn', 'onPressOut'];
+const omittedProps = ['wrapperStyle'];
 
 const PressableWithFeedbackPropTypes = {
     ...GenericPressablePropTypes.pressablePropTypes,
@@ -55,6 +55,8 @@ const PressableWithFeedback = forwardRef((props, ref) => {
         >
             <GenericPressable
                 ref={ref}
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...rest}
                 disabled={disabled}
                 onHoverIn={() => {
                     setIsHovered(true);
@@ -85,8 +87,6 @@ const PressableWithFeedback = forwardRef((props, ref) => {
                         });
                     });
                 }}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...rest}
             >
                 {(state) => (_.isFunction(props.children) ? props.children(state) : props.children)}
             </GenericPressable>
```

Comment (Skalakid) [reply src/components/Pressable/PressableWithFeedback.js:1]: @s77rt I will handle this and create PR with the follow-up

Comment (jasperhuangg) [reply src/components/Pressable/PressableWithFeedback.js:1]: @Skalakid Just wanted to let you know that this proposal created a regression: https://github.com/Expensify/App/issues/21354

Comment (s77rt) [reply src/components/Pressable/PressableWithFeedback.js:1]: Hmm wasn't this a bug even before this PR? This PR hide some bugs (and some features) we had fix that in https://github.com/Expensify/App/pull/21310

**Pull request merged**